File: geo-target/src/main/java/ru/cpb9/geotarget/ModelSourceGenerator.java
package ru.cpb9.geotarget;

import org.jetbrains.annotations.NotNull;
import ru.mipt.acsl.decode.java.generator.JavaDecodeSourcesGenerator;
import ru.mipt.acsl.decode.java.generator.JavaDecodeSourcesGeneratorConfiguration;

import java.io.File;

/**
 * @author Artem Shein
 */
public final class ModelSourceGenerator
{
    public static void main(String[] args)
    {
        JavaDecodeSourcesGeneratorConfiguration config = new JavaDecodeSourcesGeneratorConfiguration(new File("."), ModelRegistry.getRegistry());
        generate(config);
    }

    public static void generate(@NotNull JavaDecodeSourcesGeneratorConfiguration config)
    {
        JavaDecodeSourcesGenerator generator = new JavaDecodeSourcesGenerator(config);
        generator.generate();
    }

    private ModelSourceGenerator()
    {

    }
}

File: geo-target/src/main/scala/ru/mipt/acsl/geotarget/ModelSourceGenerator.scala
package ru.mipt.acsl.geotarget

import java.io.File

import ru.mipt.acsl.decode.java.generator.{JavaDecodeSourcesGenerator, JavaDecodeSourcesGeneratorConfiguration}

/**
  * @author Artem Shein
  */
object ModelSourceGenerator {
  def main(args: Array[String]): Unit = {
    val config = new JavaDecodeSourcesGeneratorConfiguration(new File("."), ModelRegistry.registry)
    generate(config)
  }
  def generate(config: JavaDecodeSourcesGeneratorConfiguration): Unit = {
    val generator = new JavaDecodeSourcesGenerator(config)
    generator.generate()
  }
}
